src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -138,24 +140,28 @@ public int getValue(){
     private static final String R_SCRIPT = "insertSizeHistogram.R";
 
     @Override
-    protected void runTool(final JavaSparkContext ctx) {
+    public boolean requiresReads(){return true;}
+
+    @Override
+    public void runTool(final JavaSparkContext ctx) {
 
         final JavaRDD<GATKRead> filteredReads = getReads();
 
+        final SAMFileHeader readsHeader = getHeaderForReads();
+        final String inputFileName = getReadSourceName();
+
         // Class where real metric-collection work is delegated to.
         final InsertSizeMetricsCollectorSpark collector = new InsertSizeMetricsCollectorSpark(filteredReads,
-                                                                                              // METRIC_ACCUMULATION_LEVEL,
-                                                                                              getHeaderForReads().getReadGroups(),
+                                                                                              readsHeader,
+                                                                                              METRIC_ACCUMULATION_LEVEL,
                                                                                               DEVIATIONS_TOL);
 
-        final SAMFileHeader readsHeader = getHeaderForReads();
-        final String inputFileName = getReadSourceName();
-
+        // similar to Picard::CollectInsertSizeMetrics.finish()
         if(OUTPUT != null) {
-            writeMetricsFile(collector, readsHeader, inputFileName);
```

Comment:
Should some validation occur to make sure that there's an output? Seems like we shouldn't process the reads at all if nothing is going to be done with the results, so this check could occur earlier.